app/services/DynamoChannelTests.scala

```diff
@@ -77,6 +77,25 @@ class DynamoChannelTests(stage: String, client: DynamoDbClient) extends StrictLo
       ).items
     }.mapError(DynamoGetError)
 
+  def getAllArchived(channel: Channel): ZIO[ZEnv, DynamoGetError, java.util.List[java.util.Map[String, AttributeValue]]] =
+    effectBlocking {
+      client.query(
+        QueryRequest
+          .builder
+          .tableName(tableName)
+          .keyConditionExpression("channel = :channel")
+          .expressionAttributeValues(Map(
+            ":channel" -> AttributeValue.builder.s(channel.toString).build,
+            ":archived" -> AttributeValue.builder.s("Archived").build
+          ).asJava)
+          .expressionAttributeNames(Map(
+            "#status" -> "status"
+          ).asJava)
+          .filterExpression("#status = :archived")
```

Comment:
What's the interaction between this line and the `expressionAttributeValues` call above?

```scala
          .expressionAttributeValues(Map(
            ":channel" -> AttributeValue.builder.s(channel.toString).build,
            ":archived" -> AttributeValue.builder.s("Archived").build
          ).asJava)
```

Reply:
It's not the clearest api!

`expressionAttributeValues` is how you define the values you'd like to reference in the query expressions. So here we define `:archived` (with value of `"Archived"`), which we can then compare against in the filter.

`expressionAttributeNames` creates an alias for the `status` field. This is only necessary because `status` is a reserved word in dynamodb, so we have to alias it to `#status`!

`filterExpression` is the filter to be applied after the query. This expression compares the aliased `status` field against the value `"Archived"`